modelling/src/main/java/org/axonframework/modelling/command/AbstractRepository.java

```diff
@@ -174,13 +176,23 @@ public Aggregate<T> loadOrCreate(@Nonnull String aggregateIdentifier, @Nonnull C
     private UnitOfWork<?> currentUnitOfWork() {
         UnitOfWork<?> uow = CurrentUnitOfWork.get();
         Class<?> messageType = uow.getMessage() != null ? uow.getMessage().getClass() : null;
-        if (messageType != null && !CommandMessage.class.isAssignableFrom(messageType)) {
-            logger.warn("The active Unit of Work is expected to contain a CommandMessage, but instead contains a [{}]",
+        if (invalidMessageType(messageType)) {
+            logger.warn("The active Unit of Work is expected to contain a CommandMessage or a DeadlineMessage, but instead contains a [{}]",
                         messageType);
         }
         return uow;
     }
 
+    private boolean invalidMessageType(Class<?> messageType) {
+        if (messageType == null) {
+            return false;
+        }
+        if (CommandMessage.class.isAssignableFrom(messageType)) {
+            return false;
+        }
+        return !Arrays.asList(messageType.getInterfaces()).contains(DeadlineMessage.class);
```

Comment:
Why can't we do an `isAssignableFrom` check as well?

Reply:
Because it's an interface, alternatively, we could do the `isAssignableFrom` on the `GenericDeadlineMessage` but then it would not work if a custom deadline implementation was used.

Reply:
Simplified using `isAssignableFrom`